cluster/gce/trusty/configure.sh

```diff
@@ -22,7 +22,8 @@
 
 config_hostname() {
   # Set the hostname to the short version.
-  short_hostname=$(hostname -s)
+  host_name=$(curl --silent "http://metadata.google.internal/computeMetadata/v1/instance/hostname" -H "Metadata-Flavor: Google")
```

Comment:
Yes, the same logic. We used to get the short name from "hostname -s". But recently we are experimenting systemd and notice systemd has a bug that when fullname is too long systemd will reset it to "localhost". So, now we switch to get the fullname from gce metadata and truncate it.